torch/csrc/distributed/c10d/Ops.cpp

```diff
@@ -58,387 +58,6 @@ TORCH_LIBRARY(c10d, m) {
-  op.call(
-      tensor,
-      process_group,
-      opts.device_ids,
-      opts.timeout.count(),
-      wait_all_ranks);
-}
-
-c10::intrusive_ptr<Work> barrier(
-    const c10::intrusive_ptr<ProcessGroup>& process_group,
-    const BarrierOptions& opts) {
-  static at::Tensor tensor;
-  // TODO: if nccl was specified then use it
-  if (process_group->getBackendType() ==
-      c10d::ProcessGroup::BackendType::NCCL) {
-    // set cuda tensor
-    tensor = at::empty(
-        {1}, at::TensorOptions().device(at::DeviceType::CUDA).dtype(at::kByte));
-  } else {
-    // Default to using cpu implementation
-    tensor = at::empty(
-        {1}, at::TensorOptions().device(at::DeviceType::CPU).dtype(at::kByte));
-  }
-
-  static auto op = c10::Dispatcher::singleton()
-                       .findSchemaOrThrow("c10d::barrier", "")
-                       .typed<c10::intrusive_ptr<::c10d::Work>(
-                           at::Tensor,
-                           const c10::intrusive_ptr<::c10d::ProcessGroup>&,
-                           const std::vector<int64_t>&,
-                           int64_t)>();
-
-  return op.call(tensor, process_group, opts.device_ids, opts.timeout.count());
-}
-
-c10::intrusive_ptr<Work> send(
-    const c10::intrusive_ptr<ProcessGroup>& process_group,
-    at::TensorList tensors,
-    int64_t dstRank,
-    int64_t tag) {
-  static auto op = c10::Dispatcher::singleton()
-                       .findSchemaOrThrow("c10d::send", "")
-                       .typed<c10::intrusive_ptr<::c10d::Work>(
-                           at::TensorList,
-                           const c10::intrusive_ptr<::c10d::ProcessGroup>&,
-                           int64_t,
-                           int64_t)>();
-  return op.call(tensors, process_group, dstRank, tag);
-}
-
-c10::intrusive_ptr<Work> recv(
-    const c10::intrusive_ptr<ProcessGroup>& process_group,
-    at::TensorList tensors,
-    int64_t srcRank,
-    int64_t tag) {
-  static auto op = c10::Dispatcher::singleton()
-                       .findSchemaOrThrow("c10d::recv_", "")
-                       .typed<c10::intrusive_ptr<::c10d::Work>(
-                           at::TensorList,
-                           const c10::intrusive_ptr<::c10d::ProcessGroup>&,
-                           int64_t,
-                           int64_t)>();
-  return op.call(tensors, process_group, srcRank, tag);
-}
-
-c10::intrusive_ptr<Work> recv_any_source(
-    const c10::intrusive_ptr<ProcessGroup>& process_group,
-    at::TensorList tensors,
-    int64_t tag) {
-  static auto op = c10::Dispatcher::singleton()
-                       .findSchemaOrThrow("c10d::recv_any_source_", "")
-                       .typed<c10::intrusive_ptr<::c10d::Work>(
-                           at::TensorList,
-                           const c10::intrusive_ptr<::c10d::ProcessGroup>&,
-                           int64_t)>();
-  return op.call(tensors, process_group, tag);
-}
-
```

Comment:
After removing this code block, should we also remove the corresponding API declarations in Ops.hpp?

Reply:
Good point! Thanks